display.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"image"
+	r "reflect"
+	"strings"
+
+	"github.com/cosmos72/gomacro/imports"
+)
+
+// Support an interface similar - but not identical - to the IPython (canonical Jupyter kernel).
+// See http://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html#IPython.display.display
+// for a good overview of the support types. Note: This is missing _repr_markdown_ and _repr_javascript_.
+
+var globalReceipt *msgReceipt // ugly global variable. any alternative?
+
+const (
+	mimeTypeHTML       = "text/html"
+	mimeTypeMarkdown   = "text/markdown"
+	mimeTypeLatex      = "text/latex"
+	mimeTypeSVG        = "image/svg+xml"
+	mimeTypePNG        = "image/png"
+	mimeTypeJPEG       = "image/jpeg"
+	mimeTypeJSON       = "application/json"
+	mimeTypeJavaScript = "application/javascript"
+)
+
+// TODO handle the metadata
+
+func render2(mimeType string, data interface{}) error {
+	return render3(mimeType, fmt.Sprint(data), data)
+}
+
+func render3(mimeType string, text string, data interface{}) error {
+	receipt := globalReceipt
+	if receipt == nil {
+		return errors.New("msgReceipt is nil, cannot send display_data message")
+	}
+	return receipt.PublishDisplayData(
+		bundledMIMEData{
+			"text/plain": text,
+			mimeType:     data,
+		}, make(bundledMIMEData))
+}
+
```

Comment:
`render2` and `render3` would then return `map[string]interface{}` instead of publishing.